java/src/main/native/ai_onnxruntime_OnnxTensor.c

```diff
@@ -15,44 +15,42 @@
  */
 JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OnnxTensor_createTensor
         (JNIEnv * jniEnv, jclass jobj, jlong apiHandle, jlong allocatorHandle, jobject dataObj, jlongArray shape, jint onnxTypeJava) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
+    (void) jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
     const OrtApi* api = (const OrtApi*) apiHandle;
     OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
     // Convert type to ONNX C enum
     ONNXTensorElementDataType onnxType = convertToONNXDataFormat(onnxTypeJava);
 
     // Extract the shape information
     jboolean mkCopy;
```

Comment:
>mkCopy [](http://example.com/codeflow?start=13&length=6)

I can not find API docs on the the Array functions and what mkCopy means. Is it an input arg or output?
If it is an input, it should be initialized.
If output and it is true, does it mean there was a copy created that needs to be deallocated properly?

Reply:
It's a variable which is written to that says if the JVM made a copy of the array or not, and the array (or reference to it) is released by the call to `ReleaseLongArrayElements` below. In practice I don't use that information at all, so it can be replaced with `NULL` like it is elsewhere in the codebase. The JNI functions are documented here - https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html.